packages/core/plugin/src/withSentryAndroid.ts

```diff
@@ -49,3 +56,50 @@ export function modifyAppBuildGradle(buildGradle: string): string {
 
   return buildGradle.replace(pattern, match => `${applyFrom}\n\n${match}`);
 }
+
+export function modifyMainApplication(config: ExpoConfig): ExpoConfig {
+  return withMainApplication(config, async config => {
+    if (!config.modResults || !config.modResults.path) {
+      warnOnce('Skipping MainApplication modification because the file does not exist.');
+      return config;
+    }
+
+    const fileName = config.modResults.path.split('/').pop();
+
+    if (config.modResults.contents.includes('RNSentrySDK.init')) {
+      warnOnce(`Your '${fileName}' already contains 'RNSentrySDK.init'.`);
+      return config;
+    }
+
+    if (config.modResults.language === 'java') {
+      if (!config.modResults.contents.includes('import io.sentry.react.RNSentrySDK;')) {
+        // Insert import statement after package declaration
+        config.modResults.contents = config.modResults.contents.replace(
+          /(package .*;\n\n?)/,
+          `$1import io.sentry.react.RNSentrySDK;\n`,
+        );
+      }
+      // Add RNSentrySDK.init
+      config.modResults.contents = config.modResults.contents.replace(
+        /(super\.onCreate\(\)[;\n]*)([ \t]*)/,
+        `$1\n$2RNSentrySDK.init(this);\n$2`,
+      );
+    } else {
+      // Kotlin
+      if (!config.modResults.contents.includes('import io.sentry.react.RNSentrySDK')) {
+        // Insert import statement after package declaration
+        config.modResults.contents = config.modResults.contents.replace(
+          /(package .*\n\n?)/,
+          `$1import io.sentry.react.RNSentrySDK\n`,
+        );
+      }
+      // Add RNSentrySDK.init
+      config.modResults.contents = config.modResults.contents.replace(
+        /(super\.onCreate\(\)[;\n]*)([ \t]*)/,
+        `$1\n$2RNSentrySDK.init(this)\n$2`,
```

Comment:
q: I think we should warn the user if `RNSentrySDK.init` wasn't inserted, to avoid giving the impression the SDK was set in some edge case where it wasn't

Reply:
Good catch @lucas-zimerman 🙇 
I've added warning for both platforms with d25db305aaed11e2a8d5e9657ffae4fcc5287eee